### ISSUE TITLE
[Feature]: Resolve aliased imports to real file paths before applying FSD layer and slice rules

### ISSUE THREAD

**Issue by Luzefiru** [effozen/eslint-plugin-fsd-lint#26]
## Is Your Feature Request Related to a Problem?

Yes.

I am using the FSD plugin in a codebase that relies on path aliases, and I am hitting a false positive when importing between files that belong to the same slice.

For example, I have a file inside the `articles` page slice importing another file from that same `articles` page slice, but the plugin still reports it as an invalid `pages` to `pages` import.

The current behavior appears to classify the import from the alias prefix instead of the real resolved file path.

Example:

```tsx
// apps/web/src/pages/articles/ui/articles-pending-page.tsx
import { type ArticleSectionKey, articleSections } from "@/pages/articles/api/queries";
import { ArticlesLayout } from "@/pages/articles/ui/articles-page";
```

Both imports stay inside the same `articles` slice in the `pages` layer, so this should be allowed.

## Describe the Solution You’d Like

I would like the plugin to resolve the actual filesystem path of an import before enforcing FSD layer and slice restrictions.

Desired behavior:

- Resolve aliased imports such as `@/pages/...` using the project alias configuration.
- Determine the actual layer and slice of both the importing file and the imported file after resolution.
- Allow imports when both files belong to the same slice, even if the import is written with an alias.
- Keep reporting violations for true cross-slice imports or upward imports across layers.

Expected interpretation for the example above:

```tsx
// importing file
apps/web/src/pages/articles/ui/articles-pending-page.tsx

// imported file
apps/web/src/pages/articles/ui/articles-page.tsx
```

These are both part of the same `pages/articles` slice, so the import should pass.

## Describe Alternatives You’ve Considered

I have considered a few workarounds, but they are all weaker than fixing the rule itself:

- Switching same-slice imports to relative paths.
- Moving reusable code into a different segment only to satisfy the lint rule.
- Disabling the rule for specific lines or files.
- Re-exporting through a public API only to avoid the false positive.

For example, this works as a workaround:

```tsx
import { ArticlesLayout } from "./articles-page";
```

But it does not solve the actual problem, because the aliased import still points to the same slice and should be valid.

## Additional Information

The expected FSD behavior here is that internal imports within the same slice should be valid regardless of whether the import is written as a relative path or as an alias.

The issue seems to be that the plugin is using the raw import string as the main signal instead of the resolved target path.

If the rule resolved imports first, it could correctly distinguish between:

- same-slice internal imports, which should be allowed
- cross-slice imports in the same layer, which may be restricted
- upward imports across layers, which should still be blocked

Example of the kind of logic I am expecting:

```tsx
// pseudo-example
const fromFile = "apps/web/src/pages/articles/ui/articles-pending-page.tsx";
const importedPath = "@/pages/articles/ui/articles-page";
const resolvedImport = "apps/web/src/pages/articles/ui/articles-page.tsx";

// expected plugin behavior
// both files belong to pages/articles
// therefore this import should be allowed
```

This would make the plugin much more reliable in projects that use TypeScript path aliases or bundler aliases heavily.

**Comment by effozen** [comment]
Thank you for issue report. I'll check it and will be inform you soon! Thanks!

**Comment by effozen** [comment]
https://github.com/effozen/eslint-plugin-fsd-lint/releases/tag/v1.2.0

Fixed and deploy.
Thank you for reporting! 
Have a good day!

**Comment by Luzefiru** [comment]
I can confirm that it's fixed and correctly detects absolute imports in the same slice. Thank you so much for making the best FSD ESLint plugin! 💪

**Comment by effozen** [comment]
I'm glad to hear that. Happy I could help 😁. Hope you have a great day!
